src/driver/alloy.rs

```diff
@@ -0,0 +1,223 @@
+            let mut trevm = trevm;
+
+            for raw_tx in self.txs.clone().into_iter() {
+                let tx = TxEnvelope::decode_2718(&mut raw_tx.to_vec().as_slice());
+
+                let tx = match tx {
+                    Ok(tx) => tx,
+                    Err(e) => return Err(trevm.errored(BundleError::TransactionDecodingError(e))),
+                };
+
+                let run_result = trevm.run_tx(&tx);
+
+                match run_result {
+                    // return immediately if errored
+                    Err(e) => {
+                        return Err(e.map_err(|e| BundleError::EVMError { inner: e }));
+                    }
+                    // Accept the state, and move on
+                    Ok(res) => match res.result() {
+                        ExecutionResult::Revert { .. } | ExecutionResult::Halt { .. } => {
+                            return Err(res.errored(BundleError::BundleReverted));
+                        }
+                        ExecutionResult::Success { .. } => {
+                            trevm = res.accept_state();
+                        }
+                    },
+                }
+            }
+
+            // return the final state
+            Ok(trevm)
+        });
+
+        match run_result {
+            Ok(trevm) => Ok(trevm),
+            Err(e) => Err(e),
+        }
+    }
+
+    fn post_bundle<Db: revm::Database + revm::DatabaseCommit>(
+        &mut self,
+        _trevm: &crate::EvmNeedsTx<'_, Ext, Db>,
+    ) -> Result<(), Self::Error<Db>> {
+        Ok(())
+    }
+}
+
+impl<Ext> BundleDriver<Ext> for EthSendBundle {
+    type Error<Db: revm::Database> = BundleError<Db>;
+
+    fn run_bundle<'a, Db: revm::Database + revm::DatabaseCommit>(
+        &mut self,
+        trevm: crate::EvmNeedsTx<'a, Ext, Db>,
+    ) -> DriveBundleResult<'a, Ext, Db, Self> {
+        // 1. Check if the block we're in is valid for this bundle. Both must match
+        if trevm.inner().block().number.to::<u64>() != self.block_number {
+            return Err(trevm.errored(BundleError::BlockNumberMismatch));
+        }
+
+        // 2. Check for start timestamp range validity
+        if let Some(min_timestamp) = self.min_timestamp {
+            if trevm.inner().block().timestamp.to::<u64>() < min_timestamp {
+                return Err(trevm.errored(BundleError::TimestampOutOfRange));
+            }
+        }
+
+        // 3. Check for end timestamp range validity
+        if let Some(max_timestamp) = self.max_timestamp {
+            if trevm.inner().block().timestamp.to::<u64>() > max_timestamp {
+                return Err(trevm.errored(BundleError::TimestampOutOfRange));
+            }
+        }
+
+        let mut t = trevm;
+
+        for raw_tx in self.txs.clone().into_iter() {
+            // Decode the transaction
+            let tx = TxEnvelope::decode_2718(&mut raw_tx.to_vec().as_slice());
+
+            let tx = match tx {
```

Comment:
could be a `map_err()?` and should we dry this up with the deser in the call function?